src/PrestaShopBundle/Translation/PrestaShopTranslatorTrait.php

```diff
@@ -53,27 +53,38 @@ public function trans($id, array $parameters = [], $domain = null, $locale = nul
             $legacy = $parameters['legacy'];
             unset($parameters['legacy']);
         }
+        $emptyParams = empty($parameters);
+        $isSprintf = $this->isSprintfString($id);
 
         if (empty($locale)) {
             $locale = null;
         }
 
-        $translated = parent::trans($id, [], $this->normalizeDomain($domain), $locale);
+        if (!$emptyParams && $isSprintf) {
+            $translated = parent::trans(
+                $id,
+                 [],
+                $this->normalizeDomain($domain),
+                $locale
+            );
+        } else {
+            $translated = parent::trans(
+                $id,
+                $parameters,
+                $this->normalizeDomain($domain),
+                $locale
+            );
+        }
 
-        // @todo to remove after the legacy translation system has ben phased out
         if ($this->shouldFallbackToLegacyModuleTranslation($id, $domain, $translated)) {
             return $this->translateUsingLegacySystem($id, $parameters, $domain, $locale);
         }
 
-        if (isset($legacy) && 'htmlspecialchars' === $legacy) {
-            $translated = call_user_func($legacy, $translated, ENT_NOQUOTES);
-        } elseif (isset($legacy)) {
-            $translated = call_user_func($legacy, $translated);
-        }
-
-        if (!empty($parameters) && $this->isSprintfString($id)) {
+        $translated = isset($legacy) ? $this->replaceHtmlSpecialChars($translated, $legacy) : $translated;
+        
+        if (!$emptyParams && $isSprintf) {
             $translated = vsprintf($translated, $parameters);
-        } elseif (!empty($parameters)) {
+        } elseif (!$emptyParams) {
```

Comment:
This elseif could be removed, because the `trans` method already perform an `strtr` internally